loop.go

```diff
@@ -0,0 +1,17 @@
+package main
+
+import (
+	"time"
+)
+
+func FetchLoop(state *State, interval int) {
+	for {
+		select {
+		case <-threadDone:
+			return
+		default:
```

Comment:
Replace it with `case <-time.After(time.Duration(interval) * time.Second)`. Otherwise, even if `threadDone` returns something, this loop will be blocked. 

You probably wanted to trigger the logic immediately and my suggestion will make it wait `interval` seconds. I think it would be better to just call `state.Fetch` before `for` loop if that was the goal.

Reply:
I was wondering about that, you're right, putting `state.Fetch()` before the loop is much better.